lms/djangoapps/course_home_api/progress/v1/serializers.py

```diff
@@ -51,11 +51,46 @@ class ChapterSerializer(serializers.Serializer):
     subsections = SubsectionSerializer(source='sections', many=True)
 
 
+class CertificateDataSerializer(serializers.Serializer):
+    cert_web_view_url = serializers.CharField()
+    download_url = serializers.CharField()
+    is_downloadable = serializers.SerializerMethodField()
+    is_requestable = serializers.SerializerMethodField()
+    msg = serializers.CharField()
+    request_cert_url = serializers.SerializerMethodField()
+    title = serializers.CharField()
+
+    def get_is_downloadable(self, cert_data):
+        return cert_data.cert_status == CertificateStatuses.downloadable and cert_data.download_url is not None
+
+    def get_is_requestable(self, cert_data):
+        return cert_data.cert_status == CertificateStatuses.requesting and cert_data.request_cert_url is not None
+
+    def get_request_cert_url(self, cert_data):
+        return reverse('generate_user_cert', args=[self.context['course_key']])
+
+
+class CreditCourseRequirementsSerializer(serializers.Serializer):
+    """
+    Serializer for credit_course_requirements
+    """
+    display_name = serializers.CharField()
+    namespace = serializers.CharField()
+    min_grade = serializers.SerializerMethodField()
+    status = serializers.CharField()
+    status_date = serializers.DateTimeField()
+
+    def get_min_grade(self, req):
+        return req['criteria']['min_grade'] * 100
+
+
 class ProgressTabSerializer(serializers.Serializer):
     """
     Serializer for progress tab
     """
-    course_blocks = CourseBlockSerializer()
```

Comment:
This says you're deleting the CourseBlockSerializer now? That doesn't seem right.

Reply:
yeah i think i don't need them, i have the assignment information from the courseware_summary and there's a more specific function call to get just the grading of the course than to look through the course blocks!